src/commands/service/create-revision.ts

```diff
@@ -38,7 +38,11 @@ export const createServiceRevisionCommandDecorator = (
     .option("")
     .action(async opts => {
       try {
-        const { orgName, personalAccessToken } = opts;
+        const { azure_devops } = Config();
+        const {
+          orgName = azure_devops && azure_devops.org,
+          personalAccessToken = azure_devops && azure_devops.access_token
+        } = opts;
```

Comment:
will this need to be performed in all these other places?

```
types.d.ts
101:  azure_devops?: {

lib/git/azure.ts
58:      personalAccessToken = config.azure_devops &&
59:        config.azure_devops.access_token,
60:      orgName = config.azure_devops && config.azure_devops.org
66:        `Unable to parse Azure DevOps Personal Access Token (azure_devops.personal_access_token) from spk config`
72:        `Unable to parse Azure DevOps Organization URL (azure_devops.devops_org_url) from spk config`

commands/mocks/spk-config.yaml
2:azure_devops: # Should this live under the `spk` configuration?

commands/hld/pipeline.ts
35:      if (!config.azure_devops) {
41:      const orgName = config.azure_devops.org!;
42:      const pat = config.azure_devops.access_token!;
43:      const hldRepo = config.azure_devops.hld_repository!;
44:      const project = config.azure_devops.project!;
45:      const manifestRepo = config.azure_devops.manifest_repository!;

commands/deployment/get.ts
159:    !config.azure_devops ||
161:    !config.azure_devops.org ||
162:    !config.azure_devops.project ||
174:    config.azure_devops.org,
175:    config.azure_devops.project,
177:    config.azure_devops.access_token
180:    config.azure_devops.org,
181:    config.azure_devops.project,
183:    config.azure_devops.access_token
186:    config.azure_devops.org,
187:    config.azure_devops.project,
189:    config.azure_devops.access_token

commands/deployment/dashboard.ts
22:        !config.azure_devops ||
23:        !config.azure_devops.org ||
24:        !config.azure_devops.project ||
72:  envVars.push("REACT_APP_PIPELINE_ORG=" + config.azure_devops!.org!);
74:  envVars.push("REACT_APP_PIPELINE_PROJECT=" + config.azure_devops!.project!);
93:  if (config.azure_devops!.access_token) {
96:      "REACT_APP_PIPELINE_ACCESS_TOKEN=" + config.azure_devops!.access_token

commands/deployment/validate.ts
46:    if (!config.azure_devops) {
47:      missingConfig.push("config.azure_devops");
49:      if (!config.azure_devops.org) {
50:        missingConfig.push("config.azure_devops.org");
52:      if (!config.azure_devops.project) {
53:        missingConfig.push("config.azure_devops.project");

commands/service/create-revision.ts
32:      "Personal access token associated with your Azure DevOps token; falls back to azure_devops.access_token in your spk config"
36:      "Your Azure DevOps organization name; falls back to azure_devops.org in your spk config"
```

Reply:
You would need to do this any place you want to use the config.
Most places throughout the code base are already doing so correctly; The error this PR fixes was because I didn't do a fallback config check on the commander side before doing a typecheck of `typeof pat !== 'string'` (it was `undefined` and didn't get the default from `Config()`)